cmd/thanos/main.go

```diff
@@ -349,13 +349,29 @@ func scheduleHTTPServer(g *run.Group, logger log.Logger, reg *prometheus.Registr
 		return errors.Wrap(err, "listen metrics address")
 	}
 
-	g.Add(func() error {
-		level.Info(logger).Log("msg", "listening for requests and metrics", "component", comp.String(), "address", httpBindAddr)
+	srv := http.Server{Handler: mux}
```

Comment:
Hm, why this and not the opposite, which is taking `InitialSync` into `g.Add` group? Essentially this breaks convention - spins up extra go routine out of run group which is not easy to handle/cancel etc. 

What's the blocker of moving `InitialSync` instead? :thinking:

Reply:
Nothing, I believe. I just didn't understand what were you saying in your first comment, now I got it. I'll try to move `InitialSync` to the rungroup.

Maybe one argument could be, adding this change in here enables also the other components' probes. However, you are right it's better to stick with the conventions and avoid synchronization issues which could be introduced by this additional goroutine.